source/darwin/dispatch_queue_event_loop.c

```diff
@@ -0,0 +1,686 @@
+    aws_rw_lock_clean_up(&dispatch_loop_context->lock);
+    aws_mem_release(dispatch_loop_context->allocator, dispatch_loop_context);
+}
+
+/* On dispatch event loop ref-count reaches 0 */
+static void s_dispatch_event_loop_destroy(void *context) {
+    // release dispatch loop
+    struct aws_event_loop *event_loop = context;
+    struct dispatch_loop *dispatch_loop = event_loop->impl_data;
+
+    if (dispatch_loop->context) {
+        // Null out the dispatch queue loop context
+        s_wlock_dispatch_loop_context(dispatch_loop->context);
+        dispatch_loop->context->io_dispatch_loop = NULL;
+        s_wunlock_dispatch_loop_context(dispatch_loop->context);
+        s_release_dispatch_loop_context(dispatch_loop->context);
+    }
+
+    // The scheduler should be cleaned up and zero out in event loop destroy task. Double check here in case the destroy
+    // function is not called or initialize was failed.
+    if (aws_task_scheduler_is_valid(&dispatch_loop->scheduler)) {
+        aws_task_scheduler_clean_up(&dispatch_loop->scheduler);
+    }
+
+    aws_mutex_clean_up(&dispatch_loop->synced_data.lock);
+    aws_mem_release(dispatch_loop->allocator, dispatch_loop);
+    aws_event_loop_clean_up_base(event_loop);
+    aws_mem_release(event_loop->alloc, event_loop);
+
+    AWS_LOGF_DEBUG(AWS_LS_IO_EVENT_LOOP, "id=%p: Destroyed Dispatch Queue Event Loop.", (void *)event_loop);
+}
+
+static const char AWS_LITERAL_APPLE_DISPATCH_QUEUE_ID_PREFIX[] = "com.amazonaws.commonruntime.eventloop.";
+static const size_t AWS_IO_APPLE_DISPATCH_QUEUE_ID_PREFIX_LENGTH =
+    AWS_ARRAY_SIZE(AWS_LITERAL_APPLE_DISPATCH_QUEUE_ID_PREFIX);
+static const size_t AWS_IO_APPLE_DISPATCH_QUEUE_ID_LENGTH =
+    AWS_IO_APPLE_DISPATCH_QUEUE_ID_PREFIX_LENGTH + AWS_UUID_STR_LEN;
+/**
+ * Generates a unique identifier for a dispatch queue in the format "com.amazonaws.commonruntime.eventloop.<UUID>".
+ * This identifier will be stored in the provided `result` buffer.
+ */
+static void s_get_unique_dispatch_queue_id(char result[AWS_IO_APPLE_DISPATCH_QUEUE_ID_LENGTH]) {
+    struct aws_uuid uuid;
+    AWS_FATAL_ASSERT(aws_uuid_init(&uuid) == AWS_OP_SUCCESS);
+    char uuid_str[AWS_UUID_STR_LEN] = {0};
+    struct aws_byte_buf uuid_buf = aws_byte_buf_from_array(uuid_str, sizeof(uuid_str));
+    uuid_buf.len = 0;
+    aws_uuid_to_str(&uuid, &uuid_buf);
+
+    memcpy(result, AWS_LITERAL_APPLE_DISPATCH_QUEUE_ID_PREFIX, AWS_IO_APPLE_DISPATCH_QUEUE_ID_PREFIX_LENGTH);
+    memcpy(result + AWS_IO_APPLE_DISPATCH_QUEUE_ID_PREFIX_LENGTH, uuid_buf.buffer, uuid_buf.len);
+}
+
+/* Setup a dispatch_queue with a scheduler. */
+struct aws_event_loop *aws_event_loop_new_with_dispatch_queue(
+    struct aws_allocator *alloc,
+    const struct aws_event_loop_options *options) {
+    AWS_PRECONDITION(options);
+    AWS_PRECONDITION(options->clock);
+
+    struct aws_event_loop *loop = aws_mem_calloc(alloc, 1, sizeof(struct aws_event_loop));
+    struct dispatch_loop *dispatch_loop = NULL;
```

Comment:
Trivial: This can be moved below to where `aws_mem_calloc` is assigning memory to it.

Reply:
Never mind, dispatch_loop needs to be available for the NULL check in clean_up